### DISCUSSION TITLE
SSG build seems not to find /server directory routes when fetching data

### DISCUSSION THREAD

**Discussion by bozicstefan** [nuxt/nuxt#22341] [Questions] [answered]
I will jump straight to the question: 
**is SSG build supposed to find /server directory routes when fetching data?**

I wanted to host my project with a static hosting and opted for _npx nuxi generate_ nuxt3 CLI command that creates the SSG build.

I noticed that once project has been built that way, the server API routes from the /server/api/** project structure cannot be found and server console and devtools network tab report 404 error.

I initially wanted to report a bug, but decided instead to check this behaviour with awesome nuxt3 community first.

I tried implementing nitro route rules and prefetching options in nuxt config, but the results are the same.

Any idea if this is intended behaviour or maybe a bug?

Thanks in advance and best regards to the Nuxt team, contributors  and community! 

My settings are: Linux Ubuntu 22.04.2 jammy jellyfish, node 18.16.1, Nuxt 3.6.5.

Reproduction steps:
---------------------------

- Initiate a new project with npx nuxi@latest init
- remove App
- add Layouts folder, with default.vue SFC file
- add Pages folder, with index.vue SFC file
- add Api subfolder with belonging .ts file to the Server folder
- create a function that fetches the server/api data on the client side
- display data somewhere in template
- IT WORKS with _npx nuxi dev_, but when built with _npx nuxi generate_ and run with _npx serve .output/public_ the 404 error occurs.

![Screenshot from 2023-07-26 11-23-18](https://github.com/nuxt/nuxt/assets/88157789/d13fc7a1-7e9c-4f5d-91fd-140a85536e8f)
![Screenshot from 2023-07-26 11-22-53](https://github.com/nuxt/nuxt/assets/88157789/7728d350-1fa3-4a2d-8588-7e461fed9ed6)
![Screenshot from 2023-07-26 11-22-39](https://github.com/nuxt/nuxt/assets/88157789/63540814-62ed-44ba-a341-1d4f5031c519)

**Comment by TheAlexLichter** [comment, marked as answer]
When you use `nuxt generate`, there *is no server*. Results of all calls done on server-side are saved in a so-called "payload file". If you want to do more requests on the client-side, you need to somehow provide a server as `nuxt generate` will **only generate static assets**.

**Reply by bozicstefan** [reply]
@manniL Thank you very much for your answer.
